test/src/test/java/hudson/console/HyperlinkNoteTest.java

```diff
@@ -72,7 +78,20 @@ public void textWithNewlinesModelHyperlinkNote() throws Exception {
                 containsString(new ModelHyperlinkNote("", 0).extraAttributes()),
                 containsString(">" + noteTextSanitized + "</a>")));
     }
+    
+    
+    @Test
+    public void textWithSingleQuote() throws Exception {
+        FreeStyleProject upstream = r.createFreeStyleProject("upstream");
+        r.createFreeStyleProject("d0wnstr3'am");
+        upstream.getPublishersList().add(new BuildTrigger("d0wnstr3'am", Result.SUCCESS));
+        FreeStyleBuild b = r.buildAndAssertSuccess(upstream);
+        r.waitUntilNoActivity();
+        HtmlPage rsp = r.createWebClient().goTo(b.getUrl()+"console");
+        assertThat(String.valueOf(rsp.getAnchorByText("d0wnstr3'am").click().getWebResponse().getStatusCode()), containsString("200"));
```

Comment:
So you want to verify the link to the downstream job exists and that by clicking it, you are able to launch it?

Reply:
> So you want to verify the link to the downstream job exists and that by clicking it, you are able to launch it?

Right! and thats working fine on my system. But it seems that no downstream build is being triggered here.